core/deployment/src/main/java/io/quarkus/deployment/util/AsmUtil.java

```diff
@@ -698,50 +703,22 @@ private static void unbox(MethodVisitor mv, String owner, String methodName, Str
             char c = chars[i];
             switch (c) {
                 case 'Z':
-                    args.add(Type.create(DotName.createSimple("boolean"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'B':
-                    args.add(Type.create(DotName.createSimple("byte"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'C':
-                    args.add(Type.create(DotName.createSimple("char"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'D':
-                    args.add(Type.create(DotName.createSimple("double"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'F':
-                    args.add(Type.create(DotName.createSimple("float"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'I':
-                    args.add(Type.create(DotName.createSimple("int"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'J':
-                    args.add(Type.create(DotName.createSimple("long"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
-                    dimensions = 0;
-                    start = i + 1;
-                    break;
                 case 'S':
-                    args.add(Type.create(DotName.createSimple("short"),
-                            dimensions > 0 ? Kind.ARRAY : Kind.PRIMITIVE));
+                    final Type type;
+                    if (dimensions == 0) {
+                        type = Type.create(DotName.createSimple(PRIMITIVE_DESCRIPTOR_TO_PRIMITIVE_CLASS_LITERAL.get(c)),
```

Comment:
Ah but it's testing against the known `c`, so I guess that's fine

Reply:
Possible values for `c` are constrained by switch and map was populated with fixed values.
The only way to get a null values is caused by switch and map misaligning due to code modification? If you are referring to this specific case, I'll do the change tomorrow

Reply:
In this case I wonder if the Switch block is really necessary?

Reply:
I left default case just for unmanaged values to avoid mixing unwanted values with primitive values; I thought it was more clear and more similar to code in other functions.
If a single case branch for primitive and invalid values is preferred, I'll change

Reply:
I don't have any strong argument against it, but given that the possible primitive values are already declared as keys in your new Map, I find it redundant to declare in the Switch too 😉

Reply:
I think it's fine. A switch inside a loop is an idiomatic method for lexing a string.

Reply:
@Ladicek @gastaldi 
How should I proceed?

Reply:
Let's keep your changes and revisit later if needed.